vortex-scan/src/work_queue.rs

```diff
@@ -188,7 +188,8 @@ impl<T> Iterator for WorkStealingIterator<T> {
                 // `steal_work` does have the side effect of stealing work, and we only want to loop
                 // again if the result of an attempt of stealing results with `Retry`, for other cases
                 // `Empty` and `Success` there is no point in trying again
-                while self.state.num_factories_constructed.load(Relaxed) < self.state.num_factories
+                // Use Acquire ordering to ensure we see all writes from other threads
+                while self.state.num_factories_constructed.load(Acquire) < self.state.num_factories
```

Comment:
Strictly speaking this is not a bug. The worst case here is spinning longer in the loop. Even with `Relaxed` the body won't be hoisted out of the loop. That said we should in all instances really use sequential consistency unless there's a real perf reason to do otherwise.